lib/rspec/buildkite/analytics/session.rb

```diff
@@ -91,13 +67,13 @@ def disconnected(connection)
           connect
           init_write_thread
         rescue SocketConnection::HandshakeError, RejectedSubscription, TimeoutError, InitialConnectionFailure, SocketConnection::SocketError => e
-          @logger.write("failed reconnection attempt #{reconnection_count} due to #{e}")
+          RSpec::Buildkite::Analytics.logger.error("failed reconnection attempt #{reconnection_count} due to #{e}")
```

Comment:
I think this one can be a warn and the final after all retries below can be an error